src/nvim/edit.c

```diff
@@ -5278,21 +5278,23 @@ insertchar (
   } else {
     int cc;
 
+    // TODO(timeyyy): refactor  this if block away, need to manually test that 
+    // inserts still work as the tests don't catch it
```

Comment:
multibyte chars can be double-width, but that's irrelevant to the code here. The point is that a test will show that multibyte chars works even if we refactor to a single branch.